src/construction/construction.py

```diff
@@ -6,277 +6,73 @@
 from typing import Callable
 from datetime import datetime
 
+from src.staging.validation import validate_data_with_schema
 
 construction_logger = logging.getLogger(__name__)
 
 
 def run_construction(
-    main_snapshot: pd.DataFrame,
-    secondary_snapshot: pd.DataFrame,
+    snapshot_df: pd.DataFrame,
     config: dict,
     check_file_exists: Callable,
-    write_csv: Callable,
     read_csv: Callable,
     run_id: int,
 ) -> pd.DataFrame:
     """Run the construction module.
 
+    The construction module reads the construction file provided by users and
+    any non-null values are used to overwrite the snapshot dataframe. This is
+    intended for users to do ad hoc updates of data without having to provide
+    a new snapshot.
+
     Args:
-        main_snapshot (pd.DataFrame): The staged and vaildated snapshot data.
-        secondary_snapshot (pd.DataFrame): The staged and validated updated
-            snapshot data.
+        snapshot_df (pd.DataFrame): The staged and vaildated snapshot data.
         config (dict): The pipeline configuration
         check_file_exists (callable): Function to check if file exists. This will
             be the hdfs or network version depending on settings.
-        write_csv (callable): Function to write to a csv file. This will be the
-            hdfs or network version depending on settings.
         read_csv (callable): Function to read a csv file. This will be the hdfs or
             network version depending on settings.
         run_id (int): The run id for this run.
     Returns:
-        constructed_df (pd.DataFrame): As main_snapshot but with records amended
-            and added from the construction files.
+        updated_snapshot_df (pd.DataFrame): As main_snapshot but with records
+            amended and flags added to mark whether a record was constructed.
     """
-    # Skip this module if the secondary snapshot isn't loaded
-    load_updated_snapshot = config["global"]["load_updated_snapshot"]
+    # Skip this module if not needed
     load_manual_construction = config["global"]["load_manual_construction"]
     if load_manual_construction is False:
```

Comment:
I think the Boolean flag should be called run_construction, not load_manual_construction, Not  a game changer, happy to pass it.